Реализовано взаимодействие с помощью Kafka и добавлена отправка уведомления на почту

pr0maxxx opened the pull request.

pr0maxxx committed:
Реализовано взаимодействие с помощью Kafka и уведомление на почту

pr0maxxx committed:
Update KafkaTaskProducer.java

pr0maxxx committed:
Update application.yml

Alexandr056 commented on taskcrud/src/main/java/com/pr0maxx/taskcrud/service/NotificationService.java:
```diff
@@ -0,0 +1,32 @@
+
+    public void sendStatusUpdate(TaskStatusUpdateMessage message) {
+        SimpleMailMessage email = new SimpleMailMessage();
+        email.setTo("maxim.vai@yandex.ru"); // В будущем можно подтянуть по userId
```
сразу сделайте для этого настройки в yml

Alexandr056 reviewed (changes requested):
status новое поле в Task, нужна миграция и логика работы с ним в сервисе

pr0maxxx committed:
Fix: добавлен статус в Task и обновлена логика обновления

# Conflicts:
#	taskcrud/src/main/java/com/pr0maxx/taskcrud/kafka/KafkaTaskProducer.java

pr0maxxx committed:
Fix: добавлен статус в Task и обновлена логика обновления

pr0maxxx committed:
Fix: добавлен статус в Task и обновлена логика обновления

Alexandr056 commented on taskcrud/src/main/java/com/pr0maxx/taskcrud/kafka/KafkaTaskProducer.java:
```diff
@@ -0,0 +1,27 @@
+
+    public void sendStatusUpdate(TaskStatusUpdateMessage message) {
+        try {
+            log.info("Отправка сообщения в Kafka: {}", message);
```
больше похоже на debug

Alexandr056 commented on taskcrud/src/main/java/com/pr0maxx/taskcrud/kafka/MessageDeserializer.java:
```diff
@@ -0,0 +1,39 @@
+@RequiredArgsConstructor
+public class MessageDeserializer<T> extends JsonDeserializer<T> {
+
+    private String getMessage(byte[] data) {
```
неиспользуемый метод

Alexandr056 commented on taskcrud/src/main/java/com/pr0maxx/taskcrud/service/NotificationService.java:
```diff
@@ -0,0 +1,30 @@
+
+    public void sendStatusUpdate(TaskStatusUpdateMessage message) {
+        SimpleMailMessage email = new SimpleMailMessage();
+        email.setTo("maxim.vai@yandex.ru");
```
в настройки)

Alexandr056 commented on taskcrud/src/main/java/com/pr0maxx/taskcrud/model/Task.java:
```diff
@@ -19,4 +19,5 @@ public class Task {
     private String title;
     private String description;
     private Long userId;
+    private String status;
```
@Column(nullable = false)
    @Enumerated(EnumType.STRING)
    private TaskStatus status;

Alexandr056 replied:
скорее так ибо статус это всегда ограниченный, фиксированный набор значений
